xds/src/main/java/io/grpc/xds/XdsClient.java

```diff
@@ -386,4 +388,66 @@ void reportClientStats(String clusterName, String serverUri, LoadStatsStore load
    */
   void cancelClientStatsReport(String clusterName) {
   }
+
+  abstract static class XdsClientFactory {
```

Comment:
Introducing this factory class doesn't seem very necessary. We could have `RefCountedXdsClientObjectPool` in its own file and it creates `XdsClient` instance via its constructor.

Reply:
We did think about that. Then we need to pass all the necessary constructor arguments of `XdsClientIml`. We can do a refactor if it's worth doing when `XdsClientImpl` is merged and start to use it.

Reply:
Sure.